### PR TITLE
TJJ-004: Create Sales Employee API

### PR DESCRIPTION
**Pull request by dylanBkainos**

### CODE REVIEW

File: src/main/java/org/example/daos/SalesEmployeeDao.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                            +
+                            "firstName, "
+                            +
+                            "lastName, "
+                            +
+                            "salary, bankAccountNumber, nationalInsuranceNumber"
+                            +
+                            "FROM Employee INNER JOIN SalesEmployee"
+                            +
+                            "WHERE Employee.employeeId IN"
```
Comment (MatthewJKainos) [inline src/main/java/org/example/daos/SalesEmployeeDao.java:33]: Perhaps try adding spaces to the end of these lines?
And also maybe testing your code..?

File: src/main/java/org/example/daos/SalesEmployeeDao.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                            +
+                            "(SELECT SalesEmployee.employeeId"
+                            +
+                            "FROM SalesEmployee)"
+            );
+
+            while (resultSet.next()) {
+                SalesEmployee salesEmployee = new SalesEmployee(
+                        resultSet.getInt("employeeId"),
+                        resultSet.getString("firstName" + " " + "lastName"),
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/SalesEmployeeDao.java:43]: out of interest would "firstName lastName" not work?

File: src/main/java/org/example/daos/SalesEmployeeDao.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                            +
+                            "salary, bankAccountNumber, nationalInsuranceNumber"
+                            +
+                            "FROM Employee INNER JOIN SalesEmployee"
+                            +
+                            "WHERE Employee.employeeId IN"
+                            +
+                            "(SELECT SalesEmployee.employeeId"
+                            +
+                            "FROM SalesEmployee)"
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/SalesEmployeeDao.java:37]: this query isnt formnatted correctly

File: src/main/java/org/example/daos/SalesEmployeeDao.java (hunk 2 above)
Comment (MatthewJKainos) [inline src/main/java/org/example/daos/SalesEmployeeDao.java:43]: Close but not quite :)

File: src/main/java/org/example/mappers/SalesEmployeeMapper.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.example.models.SalesEmployee;
+import org.example.models.SalesEmployeeResponse;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SalesEmployeeMapper {
+
+    public static List<SalesEmployeeResponse> mapSalesEmployeeListToSalesEmployeeResponseList(
```
Comment (lukemKainos) [inline src/main/java/org/example/mappers/SalesEmployeeMapper.java:11]: maybe change this to 'toSalesEmployeeResponseList

File: src/main/java/org/example/models/SalesEmployee.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    public SalesEmployee(final int employeeId,
+                         final String name,
+                         final double salary,
+                         final String bankAccountNumber,
+                         final String nationalInsuranceNumber
+    ) {
+        super(employeeId, name, salary, bankAccountNumber,
+                nationalInsuranceNumber);
+        this.commissionRate = commissionRate;
```
Comment (lukemKainos) [inline src/main/java/org/example/models/SalesEmployee.java:15]: uhh where is this set?

**Review by MatthewJKainos — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/models/SalesEmployeeResponse.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
+package org.example.models;
+
+public class SalesEmployeeResponse {
+
+    private int employeeId;
+    private String name;
+    private double salary;
+    private String bankAccountNumber;
+    private String nationalInsuranceNumber;
```
Comment (lukemKainos) [inline src/main/java/org/example/models/SalesEmployeeResponse.java:9]: do we need the setters for these? if its a response why would we not set as final and just return the object

**Review by lukemKainos — CHANGES_REQUESTED** [review]

**Review by lukemKainos — APPROVED** [review]

**Review by MatthewJKainos — APPROVED** [review]

**Review by CForde-Kainos — APPROVED** [review]

**Pull request merged**